src/Network/PushNotify/APN.hs

```diff
@@ -604,11 +610,16 @@ sendSilentMessage s token mJwt = catchErrors $
         sendApnRaw c token mJwt message
   where message = "{\"aps\":{\"content-available\":1}}"
 
-ensureOpen :: ApnSession -> IO ()
-ensureOpen s = do
-    open <- isOpen s
```

Comment:
For backwards compat, it's nicer if we leave the existing function as an alias. The types indicate what sort of thing it is. Then we're in minor bump land. A `DEPRECATED` pragma could also point users to the `ensureSessionOpen` so that `ensureOpen` can be deleted in the next major version.